nixos/modules/services/networking/solanum.nix

```diff
@@ -0,0 +1,106 @@
+{ config, lib, pkgs, ... }:
+
+let
+  inherit (lib) mkEnableOption mkIf mkOption types;
+  inherit (pkgs) solanum;
+  cfg = config.services.solanum;
+
+  configFile = pkgs.writeText "solanum.conf" ''
+    ${cfg.config}
+  '';
+in
+
+{
+
+  ###### interface
+
+  options = {
+
+    services.solanum = {
+
+      enable = mkEnableOption "Solanum IRC daemon";
+
+      config = mkOption {
```

Comment:
No RFC42 style `settings` option? 😢

Reply:
In the long run a structured config would be nice to have, but it is quite some more work to get that going, since we don't have a formatter for that.

https://github.com/solanum-ircd/solanum/blob/main/doc/ircd.conf.example

Reply:
Back when @Infinisil wasn't so busy you probably could have "tricked" him into saying "challenge accepted!"... :wink:

Reply:
yeah, when we have a formatter we can a settings options. But for right now, just a config should be enough to start.